### PR TITLE
Build Qwt 5.2.0

### PR DESCRIPTION
**Pull request by denivyruck**
Checklist
* [X] Used a fork of the feedstock to propose changes
* [X] Bumped down the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] Ensured the license file is being packaged.

I'm in a real need of this version compiled with gcc7 if possible.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,45 +8,39 @@ package:
 source:
   fn: {{ name }}-{{ version }}.tar.bz2
   url: https://sourceforge.net/projects/qwt/files/qwt/{{ version }}/{{ name }}-{{ version }}.tar.bz2
-  sha256: 1529215329e51fc562e0009505a838f427919a18b362afff441f035b2d9b5bd9
+  md5: c60b4cfac271a41524402b3ae5197890
```
Comment (marcelotrevisani) [inline recipe/meta.yaml]: Please use sha256 instead of md5

File: recipe/meta.yaml [hunk 2]
```diff
@@ -8,45 +8,39 @@ package:
...
-    - {{ cdt('mesa-libgl-devel') }}  # [linux]
-    - {{ cdt('mesa-dri-drivers') }}  # [linux]
-    - {{ cdt('libselinux') }}  # [linux]
-    - {{ cdt('libxdamage') }}  # [linux]
-    - {{ cdt('libxfixes') }}  # [linux]
-    - {{ cdt('libxxf86vm') }}  # [linux]
+    - {{ compiler('c') }}
   host:
     - qt
+    - python
```
Comment (marcelotrevisani) [inline recipe/meta.yaml]: I think python is not necessary, you may remove it

File: recipe/meta.yaml [hunk 3]
```diff
@@ -8,45 +8,39 @@ package:
...
   run:
     - qt
+    - python
```
Comment (marcelotrevisani) [inline recipe/meta.yaml]: ditto

**Comment by marcelotrevisani** [comment]
What is the reason for this PR?

**Comment by marcelotrevisani** [comment]
@conda-forge-admin , please rerender

**Comment by jschueller** [comment]
Create a v5 branch, dont merge to master

File: recipe/meta.yaml [hunk 4]
```diff
@@ -8,45 +8,37 @@ package:
...
   build:
-    - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ cdt('mesa-libgl-devel') }}  # [linux]
-    - {{ cdt('mesa-dri-drivers') }}  # [linux]
-    - {{ cdt('libselinux') }}  # [linux]
-    - {{ cdt('libxdamage') }}  # [linux]
-    - {{ cdt('libxfixes') }}  # [linux]
-    - {{ cdt('libxxf86vm') }}  # [linux]
+    - {{ compiler('c') }}
```
Comment (marcelotrevisani) [inline recipe/meta.yaml]: Please, let the recipe equal to the master, just change the version which you want

Comment (denivyruck) [reply recipe/meta.yaml]: Ops! Will do

**Comment by jschueller** [comment]
Please reopen this PR against the qwt5 branch I just created for you

**Comment by marcelotrevisani** [comment]
It not necessary to create another PR, I already changed to merge into this branch

**Comment by marcelotrevisani** [comment]
I was expecting to see if everything will pass in the CI, just after that I was thinking to create the separated branch

**Comment by marcelotrevisani** [comment]
It seems that this version is only compatible with qt4, so, you will have to pin it in your recipe

**Comment by marcelotrevisani** [comment]
There is something hard coded which is using the system toolchain, it needs to be patched

**Comment by denivyruck** [comment]
@marcelotrevisani, @jschueller thank you for your help btw!

@marcelotrevisani seems like it's some auto-generated code: 

cd src/ && $PREFIX/bin/qmake $SRC_DIR/src/src.pro -o Makefile
cd src/ && make -f Makefile 

Shouldn't it contain g++ on linux? Mac passed

**Comment by marcelotrevisani** [comment]
Nope, conda is using cross-platform compilers. You should patch the ``qwt.prf`` and get rid of the hardcoded paths and compilers
You should use ``CXX`` for c++ compilers

**Comment by marcelotrevisani** [comment]
It is passing for ``osx`` because if you look at ``qwt.prf`` is not mention anything about ``osx`` platforms. Maybe if you delete this file that should be fine, but you need to test it

**Comment by ocefpaf** [comment]
@conda-forge-admin, please rerender

**Comment by marcelotrevisani** [comment]
@conda-forge-admin, please rerender

**Comment by marcelotrevisani** [comment]
I am going to close this PR, if you still want to work on it please ping me and we will reopen it
:)

**Pull request closed without merging**